src/NServiceBus.Core.Tests/Msmq/QueuePermissionsTests.cs

```diff
@@ -22,21 +22,39 @@ public void TestFixtureSetup()
             logOutput = new StringBuilder();
             var stringWriter = new StringWriter(logOutput);
             loggerFactory.WriteTo(stringWriter);
+
         }
 
+  
         [TearDown]
         public void TearDown()
         {
             var path = @".\private$\" + testQueueName;
-
-            if (!MessageQueue.Exists(path))
+            if (MessageQueue.Exists(path))
             {
-                return;
+                MessageQueue.Delete(path);
             }
-
-            MessageQueue.Delete(path);
+            logOutput.Clear();
         }
-
+ 
+        [Test]
+        public void Should_not_warn_if_queue_has_public_access_set_to_deny()
+        {
+            // Set up a queue with read/write access for everyone/anonymous explicitly set to DENY. 
+            var everyoneGroupName = new SecurityIdentifier(WellKnownSidType.WorldSid, null).Translate(typeof(NTAccount)).ToString();
+            var anonymousGroupName = new SecurityIdentifier(WellKnownSidType.AnonymousSid, null).Translate(typeof(NTAccount)).ToString();
+            using (var queue = MessageQueue.Create(@".\private$\" + testQueueName, false))
+            {
+                queue.SetPermissions(everyoneGroupName, MessageQueueAccessRights.GenericRead, AccessControlEntryType.Deny);
```

Comment:
There are more queue access rights do we need to check any of those?

https://msdn.microsoft.com/en-us/library/system.messaging.messagequeueaccessrights(v=vs.110).aspx